ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -920,12 +953,22 @@ static bool ShouldHandleFileOpEvent(
         PerfSample findRootSample(perfTracer, PrjFSPerfCounter_FileOp_ShouldHandle_FindVirtualizationRoot);
         
         *vnodeFsidInode = Vnode_GetFsidAndInode(vnode, context);
-        *root = VirtualizationRoot_FindForVnode(
+        
+        if (isDirectory && KAUTH_FILEOP_RENAME == action)
```

Comment:
I will have to give this some more thought and testing.  I went with the current (safer?) approach because there's no guarantee that another thread won't sneak in after `VnodeCache_InvalidateCache` and insert the vnode we're processing into the cache.

Reply:
Can you better explain that scenario?  Would it OK for subsequent threads to use the cache in that instance?

Would useVnodeCache be a more descriptive variable name than invalidateCacheEntry?

Reply:
@jeschu1 based on our offline discussion today I have split `VnodeCache_FindRootForVnode` into `VnodeCache_FindRootForVnode` and `VnodeCache_RefreshRootForVnode`, restructured the code here, and added some comments.

Please let me know if you have any suggestions on making the updated code easier to follow.